keras_nlp/models/bert/bert_tasks.py

```diff
@@ -104,6 +105,20 @@ def __init__(
         self.backbone = backbone
         self.num_classes = num_classes
 
+    def get_config(self):
+        return {
+            "backbone": keras.layers.serialize(self.backbone),
+            "num_classes": self.num_classes,
+            "name": self.name,
+            "trainable": self.trainable,
+        }
+
+    @classmethod
+    def from_config(cls, config):
+        if "backbone" in config:
+            config["backbone"] = keras.layers.deserialize(config["backbone"])
```

Comment:
How does `serialize` compare to `get_config`? What would it look like to store a `preset` for a pretrained classifier model in this system? I was imagining an in-lined `config` for the `backbone` but I'm not sure what this serialized format would look like.

Reply:
Serialize will include the symbol name in a serialization friendly manner (and I think that's pretty much it?). But in short I think that's it is the more idiomatic way to save a config for nested objects like this.

Here's some examples with model and layers:
https://github.com/keras-team/keras/blob/c269e3cd8fed713fb54d2971319df0bfe6e1bf10/keras/models/sharpness_aware_minimization.py#L147
https://github.com/keras-team/keras/blob/c269e3cd8fed713fb54d2971319df0bfe6e1bf10/keras/premade_models/wide_deep.py#L214
https://github.com/keras-team/keras/blob/c269e3cd8fed713fb54d2971319df0bfe6e1bf10/keras/layers/rnn/stacked_rnn_cells.py#L204

We also do the same for serializing our activations and initializers in a config (everything hits `serialize_keras_object` eventually).

Reply:
See #377 for some of the nested configs I was imagining. For example, how would you propose we do the following?

```python
classifier_checkpoints = {
    "toxic_bert": {
        "config": {
            "backbone_config": {
                "vocabulary_size": 10000, 
                "hidden_dim": 768, 
                "intermediate_dim": 3072, 
                "num_layers": 12, 
                "num_heads": 12, 
                "max_sequence_length": 512, 
                "num_segments": 2, 
                "dropout": 0.1
        },
            "num_classes": 6,
        },
        "description": (
            "A multi-headed model capable of detecting different types "
            "of toxicity like threats, obscenity, insults, and identity-based hate."
        ),
        "weights_url": "https://storage.googleapis.com/keras-nlp/models/toxic_bert.h5",
        "weights_hash": "9b2b2139f221988759ac9cdd17050b31",
    },
}
```

Reply:
I think it's the same almost! Here's how you would do that with the code in this PR.

```
classifiers = {
    "toxic_bert": {
        "config": {
            "backbone": {
                "class_name": "keras_nlp>Bert",
                "config": {
                    "vocabulary_size": 10000,
                    "num_layers": 24,
                    "num_heads": 16,
                    "hidden_dim": 768,
                    "intermediate_dim": 3072,
                    "dropout": 0.1,
                    "max_sequence_length": 512,
                },
            },
            "num_classes": 2,
        },
    },
}
```

The advantage of this of this we have pushed things onto a standard Keras mechanism. Let's say we add an initializer argument to this and core `Bert` in the future. Like of our layers, we would serialize this with `keras.initalizers.serialize`, and it would show up in the config like this:

```
'backbone': {...},
'initializer': {'class_name': 'GlorotNormal', 'config': {'seed': None}},
'num_classes': 2,
```
or
```
'backbone': {...},
'initializer': {'class_name': 'keras_nlp>BertInitializer', 'config': {...}},
'num_classes': 2,
```

We have made our mechanism for nesting Keras objects uniform with the rest of Keras. And we leave room for other symbol support in the future (maybe not a big deal for BERT in this specific case, but nice to have generally).

Note that if we are really opposed to the serialization extra bits we could choose to avoid this specifically for the `from_preset()` constructor. But for the `from_config()` constructor I think it makes sense to follow normal Keras idioms for sub object serialization.

Reply:
Thank you for the detailed explanation @mattdangerw!